File: plugins/sbt/graphql-java-codegen-sbt-plugin/sbt-codegen-example/src/main/scala/io/github/dreamylost/service/QueryResolverImpl.java
package io.github.dreamylost.service;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequest;
import io.github.dreamylost.Config;
import io.github.dreamylost.api.QueryResolver;
import io.github.dreamylost.model.Character;
import io.github.dreamylost.model.*;
import org.springframework.http.HttpMethod;

import java.net.URI;
import java.util.List;

/**
 * @author liguobin@growingio.com
 * @version 1.0, 2020/7/16
 */
public class QueryResolverImpl implements QueryResolver {

    @Override
    public List<Human> humans() throws Exception {
        HumansQueryRequest humanQueryRequest = new HumansQueryRequest();
        HumanResponseProjection humanResponseProjection = new HumanResponseProjection();
        humanResponseProjection.id().name();

        GraphQLRequest graphQLRequest = new GraphQLRequest(humanQueryRequest, humanResponseProjection);
        HumansQueryResponse result = Config.restTemplate.exchange(URI.create(Config.productUrl), HttpMethod.POST, Config.httpEntity(graphQLRequest), HumansQueryResponse.class).getBody();

        assert result != null;
        return result.humans();
    }

    @Override
    public Droid droid(String id) {
        DroidQueryRequest productByIdQueryRequest = new DroidQueryRequest();
        productByIdQueryRequest.setId(id);

        DroidResponseProjection droidResponseProjection = new DroidResponseProjection();
        droidResponseProjection.id().name();

        GraphQLRequest graphQLRequest = new GraphQLRequest(productByIdQueryRequest, droidResponseProjection);
        DroidQueryResponse result = Config.restTemplate.exchange(URI.create(Config.productUrl), HttpMethod.POST, Config.httpEntity(graphQLRequest), DroidQueryResponse.class).getBody();

        return result.droid();
    }


    @Override
    public Character hero(Episode episode) throws Exception {
        HeroQueryRequest heroQueryRequest = new HeroQueryRequest();
        heroQueryRequest.setEpisode(episode);

        CharacterResponseProjection characterResponseProjection = new CharacterResponseProjection();
        characterResponseProjection.id().name().onDroid(new DroidResponseProjection().id().name());

        GraphQLRequest graphQLRequest = new GraphQLRequest(heroQueryRequest, characterResponseProjection);
        HeroQueryResponse result = Config.restTemplate.exchange(URI.create(Config.productUrl), HttpMethod.POST, Config.httpEntity(graphQLRequest), HeroQueryResponse.class).getBody();

        assert result != null;
        return result.hero();

    }

    @Override
    public Human human(String id) throws Exception {
        HumanQueryRequest humanQueryRequest = new HumanQueryRequest();
        humanQueryRequest.setId(id);

        HumanResponseProjection humanResponseProjection = new HumanResponseProjection();
        //not support email, because start by main instance of spring
        humanResponseProjection.id().name();

        GraphQLRequest graphQLRequest = new GraphQLRequest(humanQueryRequest, humanResponseProjection);
        HumanQueryResponse result = Config.restTemplate.exchange(URI.create(Config.productUrl), HttpMethod.POST, Config.httpEntity(graphQLRequest), HumanQueryResponse.class).getBody();

        assert result != null;
        return result.human();
    }

}

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/sbt-test/graphql-codegen-sbt-plugin/example-client-scala/src/main/scala/io/github/dreamylost/service/QueryResolverImpl.scala
package io.github.dreamylost.service

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequest
import io.github.dreamylost.api.QueryResolver
import io.github.dreamylost.model._
import io.github.dreamylost.OkHttp
import io.github.dreamylost.model.EpisodeDO.EpisodeDO

import scala.concurrent.Await
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.duration.Duration

/**
 * This is scala style deserialization
 *
 * @author liguobin@growingio.com
 * @version 1.0,2020/11/27
 */
class QueryResolverImpl extends QueryResolver {

  @throws[Exception]
  def hero(episode: EpisodeDO): CharacterDO = {
    val heroQueryRequest = HeroQueryRequest()
    heroQueryRequest.setEpisode(episode)
    val characterResponseProjection = new CharacterResponseProjection().all$(1)
    val graphQLRequest = new GraphQLRequest(heroQueryRequest, characterResponseProjection)
    val retFuture = OkHttp.executeRequest[HeroQueryResponse](graphQLRequest)
    val ret = Await.result(retFuture, Duration.Inf)
    ret.hero()
  }

  @throws[Exception]
  def human(id: String): HumanDO = {
    val humanQueryRequest = HumanQueryRequest()
    humanQueryRequest.setId(id)
    val humanResponseProjection = new HumanResponseProjection().all$(1)
    val graphQLRequest = new GraphQLRequest(humanQueryRequest, humanResponseProjection)
    val retFuture = OkHttp.executeRequest[HumanQueryResponse](graphQLRequest)
    val ret = Await.result(retFuture, Duration.Inf)
    ret.human()
  }

  @throws[Exception]
  def humans: Seq[HumanDO] = {
    val humanQueryRequest = HumansQueryRequest()
    val humanResponseProjection = new HumanResponseProjection().all$(1)
    val graphQLRequest = new GraphQLRequest(humanQueryRequest, humanResponseProjection)
    val retFuture = OkHttp.executeRequest[HumansQueryResponse](graphQLRequest)
    val ret = Await.result(retFuture, Duration.Inf)
    ret.humans()
  }

  @throws[Exception]
  def droid(id: String): DroidDO = {
    val productByIdQueryRequest = DroidQueryRequest()
    productByIdQueryRequest.setId(id)
    val droidResponseProjection = new DroidResponseProjection().all$(1)
    val graphQLRequest = new GraphQLRequest(productByIdQueryRequest, droidResponseProjection)
    val retFuture = OkHttp.executeRequest[DroidQueryResponse](graphQLRequest)
    val ret = Await.result(retFuture, Duration.Inf)
    ret.droid()
  }
}
